### PR TITLE
Remove Auto-renew and renew buttons from purchases page

### PR DESCRIPTION
**Pull request by andrii-lysenko**
Related to # 1201210512993648-as-1204413355670312/f

## Proposed Changes
For Akismet free we want to disable the auto-renew button for users. There are a number of ways to done this, but simple check seems the most straight forward and safe. However, if there are any other options, please let me know.

For example, the product might be a one time purchase, but it will involve updating product tables and deal with ones already activated. Also, `canAutoRenewBeTurnedOff` doesn't apply on all buttons to be removed, etc. 

## Testing Instructions
- checkout the branch locally and start it `yarn start`
- Purchase a Free Personal plan from `akismet.com`
- Navigate to `http://calypso.localhost:3000/me/purchases` and select akismet free subscription.
- Observe that there are no buttons to Renew Now and to disable auto-renew

233198716-65888c94-8577-401c-a521-46046be06e58

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by spraveenitpro** [comment]
### Functional Testing:

<details>
<summary>Before applying the patch</summary>

before

</details>

<details>
<summary>After applying the patch</summary>

after

</details>

Works as expected.

File: client/me/purchases/manage-purchase/purchase-meta-expiration.tsx [hunk 1]
```diff
@@ -56,15 +57,20 @@ function PurchaseMetaExpiration( {
...
-		purchase &&
-		JETPACK_LEGACY_PLANS.some( ( plan ) => plan === purchase.productSlug ) &&
-		! isRenewable( purchase );
 
-	if ( ! purchase || isDomainTransfer( purchase ) || purchase?.isInAppPurchase ) {
+	if (
+		! purchase ||
+		isDomainTransfer( purchase ) ||
+		purchase?.isInAppPurchase ||
+		PRODUCT_AKISMET_FREE === purchase?.productSlug
```
Comment (IanRamosC) [inline client/me/purchases/manage-purchase/purchase-meta-expiration.tsx]: Can we add the `isAkismetFreeProduct` function you created above to a helper or something, so we could reuse it here and the other places you did a similar check? There's a package called `calypso-products` that we've used for functions like that.

**Review by IanRamosC — COMMENTED** [review]
The change works as expected. I left a minor comment about the code, and I'm also not sure if there's a better way of handling that, I'd probably ask @Automattic/shilling

Just saw that @spraveenitpro already replied above. :)

**Review by IanRamosC — APPROVED** [review]
LGTM!

**Pull request merged**